### ISSUE TITLE
NightQA script error

### ISSUE THREAD

**Issue by rongpu** [desihub/desispec#2204]
The nightqa script failed with the following error
```
############################################
# tail of nightqa log nightqa-daily-20240327.log:#
############################################
INFO:night_qa.py:142:get_surveys_night_expids: for NIGHT=20240327 found 34 exposures from 32 tiles for SURVEY=main and 2 exposures from 2 tiles for SURVEY=special
INFO:night_qa.py:168:get_dark_night_expid: proctable_fn = /global/cfs/cdirs/desi/spectro/redux/daily/processing_tables/processing_table_daily-20240327.csv
Traceback (most recent call last):
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/bin/desi_night_qa", line 208, in <module>
    main()
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/bin/desi_night_qa", line 144, in main
    dark_expid = get_dark_night_expid(args.night, args.prod)
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/py/desispec/night_qa.py", line 185, in get_dark_night_expid
    expid = int(str(d["EXPID"][sel][0]).strip("|"))
ValueError: invalid literal for int() with base 10: '232598|232668|232665'
```

**Comment by akremin** [comment]
Yes that is expected due to changes in the pipeline. The cte correction is also bundled into the ccdcalib and we now correctly list the flat exposures included in those fits. The dark should be the first expid in the list, if there is more than one and the OBSTYPE is dark. So the logic looks good except we now need to parse the line for the first entry. I'll take care of this.

**Comment by araichoor** [comment]
I see, thanks.

**Comment by akremin** [comment]
I fixed the original problem in PR #2206 , but now the code is crashing at a later step:
```
Traceback (most recent call last):
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/bin/desi_night_qa", line 208, in <module>
    main()
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/bin/desi_night_qa", line 156, in main
    create_dark_pdf(outfns["dark"], args.night, args.prod, dark_expid, args.nproc, bkgsub_science_cameras=args.dark_bkgsub_science_cameras)
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/py/desispec/night_qa.py", line 525, in create_dark_pdf
    proc_expids = [int(expid.strip("|")) for expid in d["EXPID"]]
  File "/global/common/software/desi/perlmutter/desiconda/20230111-2.1.0/code/desispec/main/py/desispec/night_qa.py", line 525, in <listcomp>
    proc_expids = [int(expid.strip("|")) for expid in d["EXPID"]]
ValueError: invalid literal for int() with base 10: '232598|232668|232665'
```

@anand I'm happy to fix this too, but can you give me guidance on what this is doing? Is this the code for processing the 1200s morning dark? Or is this trying to find the 300s evening dark? Depending on what it's trying to find, I may approach the fix differently.

**Comment by araichoor** [comment]
ah crap, sorry I didn t think hard about other places in the code...

this part of the code is common for dark/morningdark (it s a piece of code to see in the preproc needs to be run or not).
I guess we d want to replace:
```
proc_expids = [int(expid.strip("|")) for expid in d["EXPID"]]
```
by something like:
```
proc_expids = []
for expid in d["EXPID"]:
   proc_expids += [int(_) for _ in expid.split("|")]
```
I didn t check the code, but you get the idea, hopefully!

**Comment by akremin** [comment]
Second issue is solved in PR #2207 . The code ran successfully at NERSC, so closing this issue.
